plutus-scb/app/Main.hs

```diff
@@ -338,68 +378,78 @@ runCliCommand minLogLevel config serviceAvailability (ForkCommands commands) =
+    traverse_ (\(e, s) -> (logInfo $ ContractInstance e (Set.toList s))) instances
+runCliCommand _ _ _ _ ReportTxHistory = do
     logInfo TransactionHistoryMsg
-    traverse_ (logInfo . render . pretty) =<< Core.txHistory @ContractExe
-runCliCommand _ _ _ (UpdateContract uuid endpoint payload) =
+    traverse_ (logInfo . TxHistoryItem) =<< Core.txHistory @ContractExe
+runCliCommand _ _ _ _ (UpdateContract uuid endpoint payload) =
     void $ Instance.callContractEndpoint @ContractExe (ContractInstanceId uuid) (getEndpointDescription endpoint) payload
-runCliCommand _ _ _ (ReportContractHistory uuid) = do
+runCliCommand _ _ _ _ (ReportContractHistory uuid) = do
     logInfo ContractHistoryMsg
     contracts <- Core.activeContractHistory @ContractExe (ContractInstanceId uuid)
-    itraverse_ (\i -> raise . logContract i) contracts
+    itraverse_ (\i -> logContract i) contracts
     where
-      logContract :: Int -> ContractInstanceState ContractExe -> App ()
-      logContract index contract = logInfo $ render $ parens (pretty index) <+> pretty contract
-runCliCommand _ _ _ (ProcessContractInbox uuid) = do
+      logContract index contract = logInfo $ ContractHistoryItem index contract
+runCliCommand _ _ _ _ (ProcessContractInbox uuid) = do
     logInfo ProcessInboxMsg
     Core.processContractInbox @ContractExe (ContractInstanceId uuid)
-runCliCommand _ Config{requestProcessingConfig} _ ProcessAllContractOutboxes = do
+runCliCommand _ _ Config{requestProcessingConfig} _ ProcessAllContractOutboxes = do
     let RequestProcessingConfig{requestProcessingInterval} = requestProcessingConfig
     logInfo $ ProcessAllOutboxesMsg requestProcessingInterval
     forever $ do
         _ <- liftIO . threadDelay . fromIntegral $ toMicroseconds requestProcessingInterval
         handleError @SCBError (Core.processAllContractOutboxes @ContractExe Instance.defaultMaxIterations) (logError . ContractExeSCBError)
-runCliCommand _ _ _ PSGenerator {_outputDir} =
+runCliCommand _ _ _ _ PSGenerator {_outputDir} =
     liftIO $ PSGenerator.generate _outputDir
+runCliCommand _ _ _ _ WriteDefaultConfig{_outputFile} =
+    liftIO $ defaultConfig >>= flip CM.exportConfiguration _outputFile
 
 main :: IO ()
 main = do
-    (minLogLevel, configPath, cmd) <-
+    (minLogLevel, configPath, logConfigPath, ekg, cmd) <-
         customExecParser
             (prefs $ disambiguate <> showHelpOnEmpty <> showHelpOnError)
             (info (helper <*> versionOption <*> commandLineParser) idm)
     config <- liftIO $ decodeFileThrow configPath
-    traverse_ (EKG.forkServer "localhost") (monitoringPort <$> monitoringConfig config)
+
+    logConfig <- maybe defaultConfig loadConfig logConfigPath
+    _ <- case minLogLevel of
+            Nothing -> pure ()
+            Just ll -> CM.setMinSeverity logConfig ll
+    (trace :: Trace IO AppMsg, switchboard) <- setupTrace_ logConfig "pab"
+    let trace' = natTracer (\x -> TraceLoggerT $ \_ -> x) trace
```

Comment:
Some duplication with above, perhaps?

Reply:
Fixed it in f4be403f11f04c7eced9a137ad1c24360cdcaa5c